node-scripts/src/pipeline3/impl/dockerCompose.js

```diff
@@ -0,0 +1,213 @@
+          path
+              .resolve(
+                  instanceDef.deployment.hostDir,
+                  instanceDef.name
+              ).toString()
+      );
+
+      return script;
+    },
+    getDataScript: function(instanceDef) {
+      return "";
+    },
+    getArtifactsScript: function(instanceDef) {
+      return "";
+    }
+  },
+  startInstance: {
+    getDeploymentScript: function(instanceDef) {
+      let script = "";
+
+      script += scripts.remote(
+        instanceDef.deployment.host.value,
+        "cd " +
+          path
+            .resolve(
+              instanceDef.deployment.hostDir,
+              instanceDef.name,
+              "bahmni_docker"
+            )
+            .toString() +
+          " && " +
+          "docker-compose -p " +
+          instanceDef.name +
+          " --env-file=" +
+          path
+            .resolve(
+              instanceDef.deployment.hostDir,
+              instanceDef.name,
+              instanceDef.name + ".env"
+            )
+            .toString() +
+          " up -d\n"
+      );
+      return script;
+    },
+    getDataScript: function(instanceDef) {
+      return "";
+    },
+    getArtifactsScript: function(instanceDef) {
+      return "";
+    }
+  },
+  /**
+       * Util function that wraps the passed commands so each is applied either accordingly.
+       *
+       * @param {String} containerName - The name of the container.
+       * @param {String} ifExistsCommand - The command that should run if the container exists.
+       * @param {String} elseCommand - The command that will run if the container does *not* exist.
+       *
+       * @return {String} The script as a string.
+       */
+  ifExists: function() {},
+  restart: function(instanceDef, sudo) {
+    let script = "";
+    let path = require("path");
+    let distPath = path.resolve(
+        instanceDef.deployment.hostDir, instanceDef.name,
+        "bahmni_docker").toString();
+    script += "cd " + distPath + " && ";
+    if (sudo) {
+      script += "sudo ";
+    }
+    script += "docker-compose -p " + instanceDef.name + " restart ";
+
+    return script + "\n";},
+  remove: function(instanceDef, sudo) {
+    let script = "";
+    let path = require("path");
+    let distPath = path.resolve(
+        instanceDef.deployment.hostDir, instanceDef.name,
+        "bahmni_docker").toString();
+    script += "cd " + distPath + " && ";
+    if (sudo) {
+      script += "sudo ";
+    }
+    script += "docker-compose down -v " + instanceDef.name;
+
+    return script + "\n";
+  },
+  pull: function() {},
+  exec: (instanceDef, command, service) => {
+    let script = "";
+    script += "set -e\n";
+    var path = require("path");
+    let distPath = path.resolve(
+        instanceDef.deployment.hostDir, instanceDef.name,
+        "bahmni_docker").toString();
+    script += "cd " + distPath  + " && ";
+    script +=
+      "docker-compose exec " + service + " /bin/bash -s <<" + heredoc_2 + "\n";
```

Comment:
@chinzou , I believe we are missing specifying of the `-p ...` option here.

Reply:
Thanks @rbuisson , Yes. this is should be added